graspy/match/sgm.py

```diff
@@ -0,0 +1,332 @@
+           E. T. Harley, D. E. Fishkind, R. J. Vogelstein, and C. E. Priebe, “Fast
+           approximate quadratic programming for graph matching,” PLOS one, vol. 10,
+           no. 4, p. e0121002, 2015.
+
+
+
+    """
+
+    def __init__(
+        self,
+        n_init=1,
+        init_method="barycenter",
+        max_iter=30,
+        shuffle_input=True,
+        eps=0.1,
+        gmp=False,
+    ):
+
+        if type(n_init) is int and n_init > 0:
+            self.n_init = n_init
+        else:
+            msg = '"n_init" must be a positive integer'
+            raise TypeError(msg)
+        if init_method == "rand":
+            self.init_method = "rand"
+        elif init_method == "barycenter":
+            self.init_method = "barycenter"
+            self.n_init = 1
+        else:
+            msg = 'Invalid "init_method" parameter string'
+            raise ValueError(msg)
+        if max_iter > 0 and type(max_iter) is int:
+            self.max_iter = max_iter
+        else:
+            msg = '"max_iter" must be a positive integer'
+            raise TypeError(msg)
+        if type(shuffle_input) is bool:
+            self.shuffle_input = shuffle_input
+        else:
+            msg = '"shuffle_input" must be a boolean'
+            raise TypeError(msg)
+        if eps > 0 and type(eps) is float:
+            self.eps = eps
+        else:
+            msg = '"eps" must be a positive float'
+            raise TypeError(msg)
+        if type(gmp) is bool:
+            self.gmp = gmp
+        else:
+            msg = '"gmp" must be a boolean'
+            raise TypeError(msg)
+
+    def fit(self, A, B, W1=[], W2=[]):
+        """
+        Fits the model with two assigned adjacency matrices
+
+        Parameters
+        ---------
+        A : 2d-array, square, positive
+            A square, positive adjacency matrix
+
+        B : 2d-array, square, positive
+            A square, positive adjacency matrix
+
+        W1 : 1d-array, shape (m , 1) where m <= n
+            An array where each entry is a node in A
+
+        W2 : 1d-array, shape (m , 1) where m <= n
+            An array where each entry is a node in B
+            The elements of W1 and W2 are seeds, creating a fixed
+            seeding of W1 -> W2
+
+        Returns
+        -------
+
+        self : returns an instance of self
+        """
+        A = check_array(A, copy=True, ensure_2d=True)
+        B = check_array(B, copy=True, ensure_2d=True)
+        W1 = column_or_1d(W1)
+        W2 = column_or_1d(W2)
+
+        if A.shape[0] != B.shape[0]:
+            msg = "Adjacency matrices must be of equal size"
+            raise ValueError(msg)
+        elif A.shape[0] != A.shape[1] or B.shape[0] != B.shape[1]:
+            msg = "Adjacency matrix entries must be square"
+            raise ValueError(msg)
+        elif (A >= 0).all() == False or (B >= 0).all() == False:
+            msg = "Adjacency matrix entries must be greater than or equal to zero"
```

Comment:
does Fishkind say this in the paper?

Reply:
@asaadeldin11?

Reply:
Do you mean that they are square or entries >=0?

Reply:
@asaadeldin11 entries greater than 0, can you show where he says that if he does?

Reply:
When fishkind defines adjacency matrices in the paper they are unweighted graphs 
<img width="441" alt="Screen Shot 2020-03-05 at 11 40 24 AM" src="https://user-images.githubusercontent.com/54996865/76003492-1c5b9e00-5ed6-11ea-8c2a-e59985080ce4.png">

Reply:
@bdpedigo Also in general adjacency matrices are typically defined as non negative. I think if there are negatives in the adjacency matrix it'll throw off the objective function and kind of skew everything toward the negative values

Reply:
so he doesn't explicitly say nonnegative but I assumed that he meant that

Reply:
So far I just download the sgm file and use it for the padding algorithm. And I need to comment this part of limitation since after padding there are some -1 in the adjacency matrix. If it is merged, I am worried that there will be some conflicts. Or am I using it the wrong way?